java/src/main/java/ai/onnxruntime/OnnxTensor.java

```diff
@@ -314,8 +315,9 @@ public static OnnxTensor createTensor(OrtEnvironment env, Object data) throws Or
   }
 
   /**
-   * Create a Tensor from a Java primitive or String multidimensional array. The shape is inferred
-   * from the array using reflection.
+   * Create a Tensor from a Java primitive, String, primitive multidimensional array or String
+   * multidimensional array. The shape is inferred from the object using reflection. The default
```

Comment:
>The shape is inferred from the object using reflection. [](http://example.com/codeflow?start=29&length=55)

I think there might be a problem with the statement. This function is about scalars. Scalar shape is usually empty, according the definition, although the length of data calculated is still 1. Is this in agreement with what is being inferred here.

Reply:
Yes it is. It infers the shape of scalars to be an empty array, which seems to be how it's represented in the native library (though I only figured that out by tracing the CreateTensorAsOrtValue function in the C API).